cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -219,29 +219,34 @@ private static CpuCapacity getCpuBasedOnRequirements(ResourceRequirements resour
         Integer limit = getResourceRequirement(resourceRequirements, ResourceRequirementType.LIMIT);
 
         if (limit != null) {
-            if (request == null || limit.intValue() == request.intValue()) {
+            if (request == null) {
                 return new CpuCapacity(CpuCapacity.milliCpuToCpu(limit));
+            } else {
+                return new CpuCapacity(CpuCapacity.milliCpuToCpu(request));
             }
+        } else if (request != null) {
+            return new CpuCapacity(CpuCapacity.milliCpuToCpu(request));
```

Comment:
I'm still missing some proper tests for this which would feed it the various options and check the results.

Reply:
Tests now added to `CruiseControlTest` class.